pkg/controller/webnotifier/observer.go

```diff
@@ -0,0 +1,53 @@
+package webnotifier
+
+import (
+	"encoding/json"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+)
+
+// Notifier represents a notification dispatcher.
+type Notifier interface {
+	Notify(topic string, message []byte) error
+}
+
+// Observer instance
+type Observer struct {
+	notifier Notifier
+}
+
+// NewObserver returns observer
+func NewObserver(notifier Notifier) *Observer {
+	return &Observer{notifier: notifier}
+}
+
+// RegisterAction registers action channel to observer events
+func (o *Observer) RegisterAction(topic string, ch <-chan service.DIDCommAction) {
+	go func() {
+		for action := range ch {
+			o.notify(topic, action)
+		}
+	}()
+}
+
+// RegisterStateMsg registers state channel to observer events
+func (o *Observer) RegisterStateMsg(topic string, ch <-chan service.StateMsg) {
+	go func() {
+		for msg := range ch {
+			o.notify(topic, msg)
+		}
+	}()
+}
+
+func (o *Observer) notify(topic string, v interface{}) {
+	src, err := json.Marshal(v)
```

Comment:
is there a followup to properly serialize `service.DIDCommAction` and `service.StateMsg` ?